docs/reference/indices/put-index-template.asciidoc

```diff
@@ -0,0 +1,335 @@
+[[indices-put-template]]
+=== Put index template API
+++++
+<titleabbrev>Put index template</titleabbrev>
+++++
+
+Creates or updates an index template.
+Index templates define <<index-modules-settings,settings>>, <<mapping,mappings>>, 
+and <<indices-aliases,aliases>> that can be applied automatically to new indices. 
+
+[source,console]
+--------------------------------------------------
+PUT /_index_template/template_1
+{
+  "index_patterns" : ["te*"],
+  "priority" : 1,
+  "template": {
+    "settings" : {
+      "number_of_shards" : 2
+    }
+  }
+}
+--------------------------------------------------
+// TESTSETUP
+
+//////////////////////////
+
+[source,console]
+--------------------------------------------------
+DELETE _index_template/template_*
+--------------------------------------------------
+// TEARDOWN
+
+//////////////////////////
+
+[[put-index-template-api-request]]
+==== {api-request-title}
+
+`PUT /_index_template/<index-template>`
+
+
+[[put-index-template-api-desc]]
+==== {api-description-title}
+
+{es} applies templates to new indices based on an
+index pattern that matches the index name.
```

Comment:
I'd use `wildcard pattern` instead of `index pattern`.

That pattern can now match the name of a data stream.